File: 0027-remove-element/0027-remove-element.go
func removeElement(nums []int, val int) int {
    s, f := 0, 0 

	for ; f < len(nums); f++ {
		if nums[s] == val && nums[f] != val {
			nums[s], nums[f] = nums[f], nums[s]
			s++
		}

		if nums[s] != val {
			s++
		}
	}

	return s
}

File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    int s = 0;

    for(int f = 0; f < numsSize; f++){
        if(nums[f] != val){
            nums[s++] = nums[f];
        }
    }

    return s;
}